kasa/smartcam/modules/listen.py

```diff
@@ -0,0 +1,154 @@
+    "tns1:RuleEngine/TamperDetector/Tamper": EventType.TAMPER_DETECTED,
+}
+
+
+class Listen(SmartCamModule):
+    """Implementation of lens mask module."""
+
+    manager: NotificationManager
+    callback: Callable[[EventType], None]
+    event_types: Iterable[EventType] | None
+    listening = False
+    site: web.TCPSite
+    runner: web.AppRunner
+    instance_id: str
+
+    async def _invoke_callback(self, event: EventType) -> None:
+        self.callback(event)
+
+    async def _handle_event(self, request: web.Request) -> web.Response:
+        content = await request.read()
+        result = self.manager.process(content)
+        for msg in result.NotificationMessage:
+            if (event := TOPIC_EVENT_TYPE.get(msg.Topic._value_1)) and (
+                not self.event_types or event in self.event_types
+            ):
+                asyncio.create_task(self._invoke_callback(event))
+        return web.Response()
+
+    async def listen(
+        self,
+        callback: Callable[[EventType], None],
+        camera_credentials: Credentials,
+        *,
+        event_types: Iterable[EventType] | None = None,
+        listen_ip: str | None = None,
+        listen_port: int | None = None,
+    ) -> None:
+        """Start listening for events."""
+        self.callback = callback
+        self.event_types = event_types
+        self.instance_id = str(uuid.uuid4())
+
+        if listen_port is None:
+            listen_port = DEFAULT_LISTEN_PORT
+
+        def subscription_lost() -> None:
+            _LOGGER.debug("Notification subscription lost for %s", self._device.host)
+            asyncio.create_task(self.stop())
+
+        wsdl = f"{os.path.dirname(onvif.__file__)}/wsdl/"
+
+        mycam = onvif.ONVIFCamera(
+            self._device.host,
+            2020,
+            camera_credentials.username,
+            camera_credentials.password,
+            wsdl,
+        )
+        await mycam.update_xaddrs()
+
+        address = await self._start_server(listen_ip, listen_port)
+
+        self.manager = await mycam.create_notification_manager(
+            address=address,
+            interval=timedelta(minutes=10),
+            subscription_lost_callback=subscription_lost,
+        )
+
+        self.listening = True
+        _LOGGER.debug("Listener started for %s", self._device.host)
+
+    async def stop(self) -> None:
+        """Stop the listener."""
+        if not self.listening:
+            _LOGGER.debug("Listener for %s already stopped", self._device.host)
+            return
+
+        _LOGGER.debug("Stopping listener for %s", self._device.host)
+        self.listening = False
+        await self.site.stop()
+        await self.runner.shutdown()
+
+    async def _get_host_ip(self) -> str:
+        def _get_host() -> str:
+            if sys.platform == "win32":
+                return socket.gethostbyname(socket.gethostname())
+            else:
+                res = check_output(["hostname", "-I"])  # noqa: S603, S607
```

Comment:
We shouldn't use `check_output` if we can avoid doing that anyhow.

Reply:
Now uses platform independent `get_ip` method.